client/web/src/marketing/SurveyToast.tsx

```diff
@@ -66,28 +108,32 @@ export const SurveyToast: React.FunctionComponent<SurveyToastProps> = ({ forceVi
         return null
     }
 
-    return (
-        <Toast
-            title="Tell us what you think"
-            subtitle={
-                <span id="survey-toast-scores">How likely is it that you would recommend Sourcegraph to a friend?</span>
-            }
-            cta={
-                <SurveyRatingRadio
-                    onChange={handleDismiss}
-                    openSurveyInNewTab={true}
-                    ariaLabelledby="survey-toast-scores"
+    switch (activeStep) {
+        case ToastSteps.useCase:
+            return (
+                <SurveyUseCaseForm
+                    handleDone={formState => {
+                        setUserFeedback(current => ({ ...current, ...formState }))
+                        handleSubmit()
+                        handleContinue()
+                    }}
+                    onDismiss={onDismiss}
                 />
-            }
-            footer={
-                <Checkbox
-                    id="survey-toast-refuse"
-                    label="Don't show this again"
-                    checked={shouldPermanentlyDismiss}
-                    onChange={event => setShouldPermanentlyDismiss(event.target.checked)}
+            )
+        case ToastSteps.thankYou:
+            return <SurveySuccess onDismiss={onDismiss} />
+        case ToastSteps.rate:
+            return (
+                <SurveyUserRatingForm
+                    onChange={score => setUserFeedback(current => ({ ...current, score }))}
+                    onDismiss={onDismiss}
+                    onContinue={handleContinue}
+                    toggleShouldPermanentlyDismiss={toggleShouldPermanentlyDismiss}
```

Comment:
```suggestion
                    toggleShouldPermanentlyDismiss={setShouldPermanentlyDismiss}
```

Should be the same, I think? And we shouldn't need the extra `toggleShouldPermanentlyDismiss` function in this file then